src/cljs/main/broadfcui/page/workspace/notebooks/notebooks.cljs

```diff
@@ -263,8 +274,17 @@
                    (fn [cluster]
                      (if (= (:status cluster) "Running")
                        (links/create-external
-                        {:href (str (config/leonardo-url-root) "/notebooks/" (:googleProject cluster) "/" (:clusterName cluster))
-                         :onClick #(utils/set-notebooks-access-token-cookie (utils/get-access-token))}
+                        {:href "javascript:;"
+                         :onClick
+                         #(utils/ajax
+                           {:url (str (leo-notebook-url cluster) "/setCookie")
+                            :headers {"Authorization" (str "Bearer " (utils/get-access-token))}
+                            :with-credentials? true
+                            :cross-domain true
+                            :on-done (fn [{:keys [success? get-parsed-response]}]
+                                       (if success?
+                                         (open-notebook-in-new-tab (leo-notebook-url cluster))
```

Comment:
Since this executes asynchronously from the user's click event, the popup window will be blocked by the user's browser.